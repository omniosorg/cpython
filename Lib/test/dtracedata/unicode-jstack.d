
python$target:::function-entry
/copyinstr(arg1)=="test_unicode_stack"/
{
    self->trace = 1;
}
python$target:::function-entry
/self->trace/
{
    printf("[x]");
    jstack();
}
python$target:::function-return
/copyinstr(arg1)=="test_unicode_stack"/
{
    self->trace = 0;
}

