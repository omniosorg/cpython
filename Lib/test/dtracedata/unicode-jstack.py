
def function_1():
    pass

def function_2():
    function_1()

def test_unicode_stack():
    def únícódé():
        function_2()
    function_1()
    únícódé()

test_unicode_stack()

