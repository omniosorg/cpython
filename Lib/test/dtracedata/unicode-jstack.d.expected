[x]
[PyFile:8(test_unicode_stack)]
[x]
[PyFile:2(function_1)]
[PyFile:8(test_unicode_stack)]
[x]
[PyFile:9(únícódé)]
[PyFile:9(test_unicode_stack)]
[x]
[PyFile:5(function_2)]
[PyFile:9(únícódé)]
[PyFile:9(test_unicode_stack)]
[x]
[PyFile:2(function_1)]
[PyFile:5(function_2)]
[PyFile:9(únícódé)]
[PyFile:9(test_unicode_stack)]

